### PR TITLE
Improve discover(): perform discovery on all NICs/IPs

### PR DESCRIPTION
**Pull request by pwt**
This PR changes the way that `discover()` determines the IP addresses to use for multicast discovery. It replaces the use of `gethostbyname()` with a comprehensive list of IPv4 addresses obtained using the `ifaddr` library. This should cover cases where the host has multiple NICs and/or multiple IP addresses associated with those NICs, and prevent some of the common problems encountered when using discovery. 

Addresses #764.

### CODE REVIEW

**Comment by gfischershaw** [comment]
Works fine on my setup with multiple addresses on an interface. Probably don't need the `None `default on `create_socket `any longer.

`def create_socket(interface_addr=None):`

**Comment by KennethNielsen** [comment]
My PC just have a single interface, so I don't how much it is worth, but I can confirm that discovery on the S1 system still works with this PR.

File: soco/discovery.py [hunk 1]
```diff
@@ -541,6 +506,34 @@ def _find_ipv4_networks(min_netmask):
     return ipv4_net_list
 
 
+def _find_ipv4_addresses():
+    """Discover all the host's IP addresses.
```
Comment (KennethNielsen) [inline soco/discovery.py:510]: Discover and return ... 
maybe

**Review by KennethNielsen — APPROVED** [review]
Looks great. Approved as is, although I should say that this sort of network code is not my strongest suit, so if you can poke someone from one of the other PRs do a second review that might be a good idea.

**Comment by pwt** [comment]
> My PC just have a single interface, so I don't how much it is worth, but I can confirm that discovery on the S1 system still works with this PR.

Thanks for testing, @KennethNielsen. It's now been tested on multihomed and aliased systems, and on Windows, Linux, and macOS. Not seeing any issues (and it will certainly address some existing limitations).

File: soco/discovery.py (hunk 1 above)
Comment (pwt) [reply soco/discovery.py]: Done.

**Pull request merged**